packages/support/src/Commands/Concerns/CanReadModelSchemas.php

```diff
@@ -107,6 +96,61 @@ protected function guessBelongsToRelationshipTitleColumnName(AbstractAsset $colu
             return 'title';
         }
 
-        return $schema->getPrimaryKey()->getColumns()[0];
+        return collect($schema->getIndexes($table))->firstWhere('primary')['columns'][0] ?? 'id';
+    }
+
+    protected function parseColumnType(array $column): array
+    {
+        $type = match ($column['type']) {
+            'tinyint(1)', 'bit' => 'boolean',
+            'varchar(max)', 'nvarchar(max)' => 'text',
+            default => null,
+        };
+
+        $type ??= match ($column['type_name']) {
+            'boolean', 'bool' => 'boolean',
+            'char', 'bpchar', 'nchar' => 'char',
+            'varchar', 'nvarchar' => 'string',
+            'integer', 'int', 'int4', 'tinyint', 'smallint', 'int2', 'mediumint', 'bigint', 'int8' => 'integer',
+            'decimal', 'numeric' => 'decimal',
+            'float', 'real', 'float4' => 'float',
+            'double', 'float8' => 'double',
+            'money', 'smallmoney' => 'money',
+            'date' => 'date',
+            'time', 'timetz' => 'time',
+            'datetime', 'datetime2', 'smalldatetime', 'datetimeoffset' => 'datetime',
+            'timestamp', 'timestamptz' => 'timestamp',
+            'text', 'tinytext', 'longtext', 'mediumtext', 'ntext' => 'text',
+            'json', 'jsonb' => 'json',
+
+            // 'binary', 'varbinary', 'bytea', 'image', 'blob', 'tinyblob', 'mediumblob', 'longblob' => 'binary',
+            // 'uuid', 'uniqueidentifier' => 'uuid',
+            // 'enum' => 'enum',
+            // 'set' => 'set',
+            // 'inet', 'cidr', 'macaddr', 'macaddr8' => 'string',
+            // 'bit', 'varbit' => 'bit',
+            // 'xml' => 'xml',
+            // 'year' => 'year',
+            // 'interval' => 'interval',
+            // 'geometry', 'geometrycollection', 'linestring', 'multilinestring', 'multipoint', 'multipolygon', 'point', 'polygon', 'box', 'circle', 'line', 'lseg', 'path' => 'geometry',
+            // 'geography' => 'geography',
+            // 'tsvector', 'tsquery' => 'text',
+            default => $column['type_name'],
+        };
+
+        $values = str_contains($column['type'], '(')
+            ? str_getcsv(Str::between($column['type'], '(', ')'), ',', "'")
+            : null;
+
+        $values = is_null($values) ? [] : match ($type) {
+            'string', 'char', 'binary' => ['length' => (int) $values[0]],
+            // 'enum', 'set' => ['values' => $values],
+            // 'float', 'decimal', 'double' => ['precision' => (int) $values[0], 'scale' => isset($values[1]) ? (int) $values[1] : null],
+            // 'datetime', 'timestamp', 'time', 'interval' => ['precision' => (int) $values[0]],
+            // 'geometry', 'geography' => ['subtype' => $values[0] ?? $column['type_name'] ?? null, 'srid' => isset($values[1]) ? (int) $values[1] : null],
```

Comment:
Related to https://github.com/filamentphp/filament/pull/11429/files#r1491482327

You don't have any usage for these, but I put them here in case you wanted to add something in future.